crates/noirc_frontend/src/hir/type_check/expr.rs

```diff
@@ -256,7 +264,9 @@ impl<'interner> TypeChecker<'interner> {
                         expr_span: span,
                     }
                 });
-                Type::Function(params, Box::new(lambda.return_type))
+
+                let function_type = Type::Function(params, Box::new(lambda.return_type));
+                Type::Closure(Box::new(function_type))
```

Comment:
This approach with adding the environment parameters to the actual function type and wrapping that in a Closure type wrapper seems a bit odd/roundabout to me. Can you explain why you took this approach?

Reply:
now using just a `Function` variant